sdk/personalizer/Azure.AI.Personalizer/src/Models/PersonalizerClient.cs

```diff
@@ -122,13 +136,25 @@ public PersonalizerClient(Uri endpoint, AzureKeyCredential credential, Personali
         /// <param name="endpoint"> Supported Cognitive Services endpoint. </param>
         /// <param name="credential"> A credential used to authenticate to an Azure Service. </param>
         /// <param name="isLocalInference"> A flag to determine whether to use local inference. </param>
+        /// <param name="interactionsSubsamplePercentage"> Percentage from (0,1] determines how much percentage of interaction events to consider </param>
+        /// <param name="observationsSubSamplePercentage"> Percentage from (0,1] determines how much percentage of observation events to consider </param>
         /// <param name="options"> The options for configuring the client. </param>
-        public PersonalizerClient(Uri endpoint, AzureKeyCredential credential, bool isLocalInference, PersonalizerClientOptions options = null) :
+        public PersonalizerClient(Uri endpoint, AzureKeyCredential credential, bool isLocalInference, float interactionsSubsamplePercentage = 100, float observationsSubSamplePercentage = 100, PersonalizerClientOptions options = null) :
             this(endpoint, credential, options)
         {
             _isLocalInference = isLocalInference;
             if (isLocalInference)
             {
+                if (0 >= interactionsSubsamplePercentage || interactionsSubsamplePercentage > 1)
```

Comment:
Can you create a private method to validate the 2 inputs to avoid the duplication?

Reply:
added private method to validate and assign two variables